### ISSUE TITLE
 Error: Rate limit exceeded - please wait! 

### ISSUE THREAD

**Issue by rdatasculptor** [mkearney/rtweet#24]
First of all: thank you very much for making and sharing this package! 

All works well except for the `search_users()` function. I have a list of names of persons of whom I want to check if they have a twitter account or not. When I use `search_users()` in a loop, it says after checking just a few, e.g. six, names:  Error: Rate limit exceeded - please wait!
Ofcourse I know the twitter api has limits, but they seem to be reached so quickly. Any thoughts about how to deal with this problem? 

kind regards,
Jelle

**Comment by mkearney** [comment]
I've updated the Github version, so this should now be fixed along with a host of other fixes and improvements. Let me know if you have any more problems with it!

```
install.packages("devtools")
devtools::install_github("mkearney/rtweet")
library(rtweet)
```

**Comment by rdatasculptor** [comment]
thank you for the quick response! Unfortenately it didn't seem to help much. After 7 iterations, meaning 7 search_users calls it says the "error Rate limit exceeded - please wait!", again.

kind regards,
jelle

**Comment by rdatasculptor** [comment]
After a closer inspection, I noticed there seem to be less search hits than with the earlier version of the function.

**Comment by rdatasculptor** [comment]
Maybe it's best to give you the code that causes the problem:

`df <- structure(list(name = structure(c(3L, 7L, 5L, 9L, 1L, 6L, 2L, 
4L, 8L), .Label = c("Amma Asante", "Farshad Bashir", "Fleur Agema", 
"Harm Beertema", "Khadija Arib", "Malik Azmani", "Mustafa Amhaouch", 
"Salima Belhaj", "Tamara van Ark"), class = "factor"), number = structure(c(1L, 
2L, 3L, 4L, 5L, 6L, 7L, 1L, 2L), .Label = c("1", "2", "3", "4", 
"5", "6", "7"), class = "factor")), .Names = c("name", "number"
), row.names = c(NA, -9L), class = "data.frame")

searchfile <- list()
for (i in 1:nrow(df)){
  search <- search_users(df[i,1],n=2)
  search$user_id <- as.numeric(search$user_id)
  searchfile[[i]] <- search 
  print(i)
}
after i == 7 it gives the error.`

**Comment by mkearney** [comment]
The problem here is that you're piping a factor from a data.frame into the query argument, which expects a character vector. When I coerced it to as character, it ran great!

```{r}
df <- structure(list(name = structure(c(3L, 7L, 5L, 9L, 1L, 6L, 2L, 
  4L, 8L), .Label = c("Amma Asante", "Farshad Bashir", "Fleur Agema", 
    "Harm Beertema", "Khadija Arib", "Malik Azmani", "Mustafa Amhaouch", 
    "Salima Belhaj", "Tamara van Ark"), class = "factor"), number = structure(c(1L, 
      2L, 3L, 4L, 5L, 6L, 7L, 1L, 2L), .Label = c("1", "2", "3", "4", 
        "5", "6", "7"), class = "factor")), .Names = c("name", "number"
        ), row.names = c(NA, -9L), class = "data.frame")

searchfile <- list()
for (i in 1:nrow(df)){
  search <- search_users(as.character(df[i,1]),n=2)
  search$user_id <- as.numeric(search$user_id)
  searchfile[[i]] <- search 
  print(i)
}
searchfile
```

**Comment by rdatasculptor** [comment]
thanks! That was so obvious that I am surprised I didn't think of that myself. 

Now it runs much faster, but it still gives the error after about 17 iterations. I guess I have to try to live with that :-)

**Comment by rdatasculptor** [comment]
One last remark. The problem was that I tried to search for a user using characters with foreign accents. The error said that the rate was exceeded when in fact it was a case of foreign accents. I used this [solution](http://stackoverflow.com/questions/20495598/replace-accented-characters-in-r-with-non-accented-counterpart-utf-8-encoding) for getting rid of the accents. Now it works like a charm.
Thanks again!

Kind regards,
Jelle
